{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda3/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:516: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint8 = np.dtype([(\"qint8\", np.int8, 1)])\n",
      "/anaconda3/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:517: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint8 = np.dtype([(\"quint8\", np.uint8, 1)])\n",
      "/anaconda3/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:518: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint16 = np.dtype([(\"qint16\", np.int16, 1)])\n",
      "/anaconda3/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint16 = np.dtype([(\"quint16\", np.uint16, 1)])\n",
      "/anaconda3/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:520: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint32 = np.dtype([(\"qint32\", np.int32, 1)])\n",
      "/anaconda3/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:525: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  np_resource = np.dtype([(\"resource\", np.ubyte, 1)])\n",
      "/anaconda3/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:541: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint8 = np.dtype([(\"qint8\", np.int8, 1)])\n",
      "/anaconda3/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:542: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint8 = np.dtype([(\"quint8\", np.uint8, 1)])\n",
      "/anaconda3/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:543: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint16 = np.dtype([(\"qint16\", np.int16, 1)])\n",
      "/anaconda3/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:544: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint16 = np.dtype([(\"quint16\", np.uint16, 1)])\n",
      "/anaconda3/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:545: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint32 = np.dtype([(\"qint32\", np.int32, 1)])\n",
      "/anaconda3/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:550: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  np_resource = np.dtype([(\"resource\", np.ubyte, 1)])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:From /anaconda3/lib/python3.7/site-packages/tensorflow/python/keras/initializers.py:143: calling RandomNormal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:From /anaconda3/lib/python3.7/site-packages/tensorflow/python/ops/init_ops.py:1251: calling VarianceScaling.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x641edd6d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x641edd6d0>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x641edd6d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x641edd6d0>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x641fa7850>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x641fa7850>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x641fa7850>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x641fa7850>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x64236d390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x64236d390>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x64236d390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x64236d390>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x6426f92d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x6426f92d0>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x6426f92d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x6426f92d0>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x642776650>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x642776650>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x642776650>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x642776650>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x64274f810>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x64274f810>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x64274f810>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x64274f810>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6428a1ad0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6428a1ad0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6428a1ad0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6428a1ad0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x642889bd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x642889bd0>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x642889bd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x642889bd0>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x6426f92d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x6426f92d0>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x6426f92d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x6426f92d0>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x641edd6d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x641edd6d0>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x641edd6d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x641edd6d0>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x642776650>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x642776650>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x642776650>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x642776650>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x641fa7850>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x641fa7850>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x641fa7850>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x641fa7850>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x64274f810>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x64274f810>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x64274f810>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x64274f810>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6428a1ad0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6428a1ad0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6428a1ad0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6428a1ad0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642463090>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6424d4690>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425d4fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c4350>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x642767dd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6425c2b50>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x64274f810>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x64274f810>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x64274f810>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x64274f810>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6428a1ad0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6428a1ad0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6428a1ad0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PixelNormalization.call of <__main__.PixelNormalization object at 0x6428a1ad0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x642889bd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x642889bd0>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x642889bd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x642889bd0>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x64236d390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x64236d390>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method _Merge.call of <__main__.WeightedSum object at 0x64236d390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method _Merge.call of <__main__.WeightedSum object at 0x64236d390>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING:tensorflow:Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n",
      "WARNING: Entity <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method MinibatchStdev.call of <__main__.MinibatchStdev object at 0x641c58c10>>: AttributeError: module 'gast' has no attribute 'Num'\n"
     ]
    },
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '../../birds2.npy'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-ba0bbc8e836d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m    475\u001b[0m \u001b[0;31m# load image data\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    476\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mnumpy\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 477\u001b[0;31m \u001b[0mdataset\u001b[0m \u001b[0;34m=\u001b[0m  \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mload\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'../../birds2.npy'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    478\u001b[0m \u001b[0;31m#load_real_samples('img_align_celeba_128.npz')\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    479\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Loaded'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdataset\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.7/site-packages/numpy/lib/npyio.py\u001b[0m in \u001b[0;36mload\u001b[0;34m(file, mmap_mode, allow_pickle, fix_imports, encoding)\u001b[0m\n\u001b[1;32m    426\u001b[0m         \u001b[0mown_fid\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    427\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 428\u001b[0;31m         \u001b[0mfid\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos_fspath\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfile\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"rb\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    429\u001b[0m         \u001b[0mown_fid\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    430\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '../../birds2.npy'"
     ]
    }
   ],
   "source": [
    "from math import sqrt\n",
    "from numpy import load\n",
    "from numpy import asarray\n",
    "from numpy import zeros\n",
    "from numpy import ones\n",
    "from numpy.random import randn\n",
    "from numpy.random import randint\n",
    "from PIL import Image\n",
    "import cv2\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras.optimizers import Adam, SGD, Adamax,Adagrad\n",
    "from tensorflow.keras.models import Sequential\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input\n",
    "from tensorflow.keras.layers import Dense\n",
    "from tensorflow.keras.layers import Flatten\n",
    "from tensorflow.keras.layers import Reshape\n",
    "from tensorflow.keras.layers import Conv2D\n",
    "from tensorflow.keras.layers import UpSampling2D\n",
    "from tensorflow.keras.layers import AveragePooling2D\n",
    "from tensorflow.keras.layers import LeakyReLU\n",
    "from tensorflow.keras.layers import Layer\n",
    "from tensorflow.keras.layers import Add\n",
    "from tensorflow.keras.constraints import max_norm\n",
    "from tensorflow.keras.initializers import RandomNormal\n",
    "from tensorflow.keras import backend\n",
    "from tensorflow.keras.models import load_model\n",
    "from matplotlib import pyplot\n",
    "\n",
    "#tf.reset_default_graph()\n",
    "class PixelNormalization(Layer):\n",
    "        # initialize the layer\n",
    "        def __init__(self, **kwargs):\n",
    "                super(PixelNormalization, self).__init__(**kwargs)\n",
    "\n",
    "        # perform the operation\n",
    "        def call(self, inputs):\n",
    "                # calculate square pixel values\n",
    "                values = inputs**2.0\n",
    "                # calculate the mean pixel values\n",
    "                mean_values = backend.mean(values, axis=-1, keepdims=True)\n",
    "                # ensure the mean is not zero\n",
    "                mean_values += 1.0e-8\n",
    "                # calculate the sqrt of the mean squared value (L2 norm)\n",
    "                l2 = backend.sqrt(mean_values)\n",
    "                # normalize values by the l2 norm\n",
    "                normalized = inputs / l2\n",
    "                return normalized\n",
    "        def get_config(self):\n",
    "            cfg = super().get_config()\n",
    "            return cfg\n",
    "        # define the output shape of the layer\n",
    "        def compute_output_shape(self, input_shape):\n",
    "                return input_shape\n",
    "\n",
    "# mini-batch standard deviation layer\n",
    "class MinibatchStdev(Layer):\n",
    "        # initialize the layer\n",
    "        def __init__(self, **kwargs):\n",
    "                super(MinibatchStdev, self).__init__(**kwargs)\n",
    "\n",
    "        def get_config(self):\n",
    "            cfg = super().get_config()\n",
    "            return cfg\n",
    "        # perform the operation\n",
    "        def call(self, inputs):\n",
    "                # calculate the mean value for each pixel across channels\n",
    "                mean = backend.mean(inputs, axis=0, keepdims=True)\n",
    "                # calculate the squared differences between pixel values and mean\n",
    "                squ_diffs = backend.square(inputs - mean)\n",
    "                # calculate the average of the squared differences (variance)\n",
    "                mean_sq_diff = backend.mean(squ_diffs, axis=0, keepdims=True)\n",
    "                # add a small value to avoid a blow-up when we calculate stdev\n",
    "                mean_sq_diff += 1e-8\n",
    "                # square root of the variance (stdev)\n",
    "                stdev = backend.sqrt(mean_sq_diff)\n",
    "                # calculate the mean standard deviation across each pixel coord\n",
    "                mean_pix = backend.mean(stdev, keepdims=True)\n",
    "                # scale this up to be the size of one input feature map for each sample\n",
    "                shape = backend.shape(inputs)\n",
    "                output = backend.tile(mean_pix, (shape[0], shape[1], shape[2], 1))\n",
    "                # concatenate with the output\n",
    "                combined = backend.concatenate([inputs, output], axis=-1)\n",
    "                return combined\n",
    "        \n",
    "        # define the output shape of the layer\n",
    "        def compute_output_shape(self, input_shape):\n",
    "                # create a copy of the input shape as a list\n",
    "                input_shape = list(input_shape)\n",
    "                # add one to the channel dimension (assume channels-last)\n",
    "                input_shape[-1] += 1\n",
    "                # convert list to a tuple\n",
    "                return tuple(input_shape)\n",
    "\n",
    "# weighted sum output\n",
    "class WeightedSum(Add):\n",
    "        # init with default value\n",
    "        def __init__(self, alpha=0.0, **kwargs):\n",
    "                super(WeightedSum, self).__init__(**kwargs)\n",
    "                self.alpha = backend.variable(alpha, name='ws_alpha')\n",
    "        def get_config(self):\n",
    "            cfg = super().get_config()\n",
    "            return cfg\n",
    "        # output a weighted sum of inputs\n",
    "        def _merge_function(self, inputs):\n",
    "                # only supports a weighted sum of two inputs\n",
    "                assert (len(inputs) == 2)\n",
    "                # ((1-a) * input1) + (a * input2)\n",
    "                output = ((1.0 - self.alpha) * inputs[0]) + (self.alpha * inputs[1])\n",
    "                return output\n",
    "\n",
    "# calculate wasserstein loss\n",
    "def wasserstein_loss(y_true, y_pred):\n",
    "        return backend.mean(y_true * y_pred)\n",
    "\n",
    "# add a discriminator block\n",
    "def add_discriminator_block(old_model, n_input_layers=3):\n",
    "        # weight initialization\n",
    "        init = RandomNormal(stddev=0.02)\n",
    "        # weight constraint\n",
    "        const = max_norm(1.0)\n",
    "        # get shape of existing model\n",
    "        in_shape = list(old_model.input.shape)\n",
    "        # define new input shape as double the size\n",
    "        input_shape = (in_shape[-2]*2, in_shape[-2]*2, in_shape[-1])\n",
    "        in_image = Input(shape=input_shape)\n",
    "        # define new input processing layer\n",
    "        d = Conv2D(128, (1,1), padding='same', kernel_initializer=init, kernel_constraint=const)(in_image)\n",
    "        d = LeakyReLU(alpha=0.2)(d)\n",
    "        # define new block\n",
    "        d = Conv2D(128, (3,3), padding='same', kernel_initializer=init, kernel_constraint=const)(d)\n",
    "        d = LeakyReLU(alpha=0.2)(d)\n",
    "        d = Conv2D(128, (3,3), padding='same', kernel_initializer=init, kernel_constraint=const)(d)\n",
    "        d = LeakyReLU(alpha=0.2)(d)\n",
    "        d = AveragePooling2D()(d)\n",
    "        block_new = d\n",
    "        # skip the input, 1x1 and activation for the old model\n",
    "        for i in range(n_input_layers, len(old_model.layers)):\n",
    "                d = old_model.layers[i](d)\n",
    "        # define straight-through model\n",
    "        model1 = Model(in_image, d)\n",
    "        # compile model\n",
    "        model1.compile(loss=wasserstein_loss, optimizer=Adam(lr=0.002, beta_1=0 , beta_2=0.9, amsgrad=True))\n",
    "        \n",
    "        \n",
    "        # downsample the new larger image\n",
    "        downsample = AveragePooling2D()(in_image)\n",
    "        # connect old input processing to downsampled new input\n",
    "        block_old = old_model.layers[1](downsample)\n",
    "        block_old = old_model.layers[2](block_old)\n",
    "        # fade in output of old model input layer with new input\n",
    "        d = WeightedSum()([block_old, block_new])\n",
    "        # skip the input, 1x1 and activation for the old model\n",
    "        for i in range(n_input_layers, len(old_model.layers)):\n",
    "                d = old_model.layers[i](d)\n",
    "        # define straight-through model\n",
    "        model2 = Model(in_image, d)\n",
    "        # compile model\n",
    "        model2.compile(loss=wasserstein_loss, optimizer=Adam(lr=0.002, beta_1=0 , beta_2=0.9, amsgrad=True))\n",
    "\n",
    "        return [model1, model2]\n",
    "\n",
    "# define the discriminator models for each image resolution\n",
    "def define_discriminator(n_blocks, input_shape=(4,4,3)):\n",
    "        # weight initialization\n",
    "        init = RandomNormal(stddev=0.02)\n",
    "        # weight constraint\n",
    "        const = max_norm(1.0)\n",
    "        model_list = list()\n",
    "        # base model input\n",
    "        in_image = Input(shape=input_shape)\n",
    "        # conv 1x1\n",
    "        d = Conv2D(128, (1,1), padding='same', kernel_initializer=init, kernel_constraint=const)(in_image)\n",
    "        d = LeakyReLU(alpha=0.2)(d)\n",
    "        # conv 3x3 (output block)\n",
    "        d = MinibatchStdev()(d)\n",
    "        d = Conv2D(128, (3,3), padding='same', kernel_initializer=init, kernel_constraint=const)(d)\n",
    "        d = LeakyReLU(alpha=0.2)(d)\n",
    "        # conv 4x4\n",
    "        d = Conv2D(128, (4,4), padding='same', kernel_initializer=init, kernel_constraint=const)(d)\n",
    "        d = LeakyReLU(alpha=0.2)(d)\n",
    "        # dense output layer\n",
    "        d = Flatten()(d)\n",
    "        out_class = Dense(1)(d)\n",
    "        # define model\n",
    "        model = Model(in_image, out_class)\n",
    "        # compile model\n",
    "        model.compile(loss=wasserstein_loss, optimizer=Adam(lr=0.002, beta_1=0 , beta_2=0.9, amsgrad=True))\n",
    "        #model.compile(loss=wasserstein_loss, optimizer=Adam(lr=0.001, beta_1=0, beta_2=0.99, epsilon=10e-8))\n",
    "\n",
    "        # store model\n",
    "        model_list.append([model, model])\n",
    "        # create submodels\n",
    "        for i in range(1, n_blocks):\n",
    "                # get prior model without the fade-on\n",
    "                old_model = model_list[i - 1][0]\n",
    "                # create new model for next resolution\n",
    "                models = add_discriminator_block(old_model)\n",
    "                # store model\n",
    "                model_list.append(models)\n",
    "        return model_list\n",
    "\n",
    "# add a generator block\n",
    "def add_generator_block(old_model):\n",
    "        # weight initialization\n",
    "        init = RandomNormal(stddev=0.02)\n",
    "        # weight constraint\n",
    "        const = max_norm(1.0)\n",
    "        # get the end of the last block\n",
    "        block_end = old_model.layers[-2].output\n",
    "        # upsample, and define new block\n",
    "        upsampling = UpSampling2D()(block_end)\n",
    "        g = Conv2D(128, (3,3), padding='same', kernel_initializer=init, kernel_constraint=const)(upsampling)\n",
    "        g = PixelNormalization()(g)\n",
    "        g = LeakyReLU(alpha=0.2)(g)\n",
    "        g = Conv2D(128, (3,3), padding='same', kernel_initializer=init, kernel_constraint=const)(g)\n",
    "        g = PixelNormalization()(g)\n",
    "        g = LeakyReLU(alpha=0.2)(g)\n",
    "        # add new output layer\n",
    "        out_image = Conv2D(3, (1,1), padding='same', kernel_initializer=init, kernel_constraint=const)(g)\n",
    "        # define model\n",
    "        model1 = Model(old_model.input, out_image)\n",
    "        # get the output layer from old model\n",
    "        out_old = old_model.layers[-1]\n",
    "        # connect the upsampling to the old output layer\n",
    "        out_image2 = out_old(upsampling)\n",
    "        # define new output image as the weighted sum of the old and new models\n",
    "        merged = WeightedSum()([out_image2, out_image])\n",
    "        # define model\n",
    "        model2 = Model(old_model.input, merged)\n",
    "        return [model1, model2]\n",
    "\n",
    "# define generator models\n",
    "def define_generator(latent_dim, n_blocks, in_dim=4):\n",
    "        # weight initialization\n",
    "        init = RandomNormal(stddev=0.02)\n",
    "        # weight constraint\n",
    "        const = max_norm(1.0)\n",
    "        model_list = list()\n",
    "        # base model latent input\n",
    "        in_latent = Input(shape=(latent_dim,))\n",
    "        # linear scale up to activation maps\n",
    "        g  = Dense(128 * in_dim * in_dim, kernel_initializer=init, kernel_constraint=const)(in_latent)\n",
    "        g = Reshape((in_dim, in_dim, 128))(g)\n",
    "        # conv 4x4, input block\n",
    "        g = Conv2D(128, (3,3), padding='same', kernel_initializer=init, kernel_constraint=const)(g)\n",
    "        g = PixelNormalization()(g)\n",
    "        g = LeakyReLU(alpha=0.2)(g)\n",
    "        # conv 3x3\n",
    "        g = Conv2D(128, (3,3), padding='same', kernel_initializer=init, kernel_constraint=const)(g)\n",
    "        g = PixelNormalization()(g)\n",
    "        g = LeakyReLU(alpha=0.2)(g)\n",
    "        # conv 1x1, output block\n",
    "        out_image = Conv2D(3, (1,1), padding='same', kernel_initializer=init, kernel_constraint=const)(g)\n",
    "        # define model\n",
    "        model = Model(in_latent, out_image)\n",
    "        # store model\n",
    "        model_list.append([model, model])\n",
    "        # create submodels\n",
    "        for i in range(1, n_blocks):\n",
    "                # get prior model without the fade-on\n",
    "                old_model = model_list[i - 1][0]\n",
    "                # create new model for next resolution\n",
    "                models = add_generator_block(old_model)\n",
    "                # store model\n",
    "                model_list.append(models)\n",
    "        return model_list\n",
    "\n",
    "# define composite models for training generators via discriminators\n",
    "def define_composite(discriminators, generators):\n",
    "        model_list = list()\n",
    "        # create composite models\n",
    "        for i in range(len(discriminators)):\n",
    "                g_models, d_models = generators[i], discriminators[i]\n",
    "                # straight-through model\n",
    "                d_models[0].trainable = False\n",
    "                model1 = Sequential()\n",
    "                model1.add(g_models[0])\n",
    "                model1.add(d_models[0])\n",
    "                model1.compile(loss=wasserstein_loss, optimizer=Adam(lr=0.002, beta_1=0 , beta_2=0.9, amsgrad=True))\n",
    "                # fade-in model\n",
    "                d_models[1].trainable = False\n",
    "                model2 = Sequential()\n",
    "                model2.add(g_models[1])\n",
    "                model2.add(d_models[1])\n",
    "                model2.compile(loss=wasserstein_loss, optimizer=Adam(lr=0.002, beta_1=0 , beta_2=0.9))\n",
    "                # store\n",
    "                model_list.append([model1, model2])\n",
    "        return model_list\n",
    "\n",
    "# load dataset\n",
    "def load_real_samples(filename):\n",
    "        # load dataset\n",
    "        data = load(filename)\n",
    "        # extract numpy array\n",
    "        X = data['arr_0']\n",
    "        # convert from ints to floats\n",
    "        X = X.astype('float32')\n",
    "        # scale from [0,255] to [-1,1]\n",
    "        X = (X - 127.5) / 127.5\n",
    "        return X\n",
    "\n",
    "# select real samples\n",
    "def generate_real_samples(dataset, n_samples):\n",
    "        # choose random instances\n",
    "        ix = randint(0, dataset.shape[0], n_samples)\n",
    "        # select images\n",
    "        X = dataset[ix]\n",
    "        # generate class labels\n",
    "        y = ones((n_samples, 1))\n",
    "        return X, y\n",
    "\n",
    "# generate points in latent space as input for the generator\n",
    "def generate_latent_points(latent_dim, n_samples):\n",
    "        # generate points in the latent space\n",
    "        x_input = randn(latent_dim * n_samples)\n",
    "        # reshape into a batch of inputs for the network\n",
    "        x_input = x_input.reshape(n_samples, latent_dim)\n",
    "        return x_input\n",
    "\n",
    "# use the generator to generate n fake examples, with class labels\n",
    "def generate_fake_samples(generator, latent_dim, n_samples):\n",
    "        # generate points in latent space\n",
    "        x_input = generate_latent_points(latent_dim, n_samples)\n",
    "        # predict outputs\n",
    "        X = generator.predict(x_input)\n",
    "        # create class labels\n",
    "        y = -ones((n_samples, 1))\n",
    "        return X, y\n",
    "\n",
    "# update the alpha value on each instance of WeightedSum\n",
    "def update_fadein(models, step, n_steps):\n",
    "        # calculate current alpha (linear from 0 to 1)\n",
    "        alpha = step / float(n_steps - 1)\n",
    "        # update the alpha for each model\n",
    "        for model in models:\n",
    "                for layer in model.layers:\n",
    "                        if isinstance(layer, WeightedSum):\n",
    "                                backend.set_value(layer.alpha, alpha)\n",
    "\n",
    "# train a generator and discriminator\n",
    "def train_epochs(g_model, d_model, gan_model, dataset, n_epochs, n_batch, fadein=False):\n",
    "        # calculate the number of batches per training epoch\n",
    "        bat_per_epo = int(dataset.shape[0] / n_batch)\n",
    "        # calculate the number of training iterations\n",
    "        n_steps = bat_per_epo * n_epochs\n",
    "        # calculate the size of half a batch of samples\n",
    "        half_batch = int(n_batch / 2)\n",
    "        # manually enumerate epochs\n",
    "        for i in range(n_steps):\n",
    "                # update alpha for all WeightedSum layers when fading in new blocks\n",
    "                if fadein:\n",
    "                        update_fadein([g_model, d_model, gan_model], i, n_steps)\n",
    "                # prepare real and fake samples\n",
    "                X_real, y_real = generate_real_samples(dataset, half_batch)\n",
    "                X_fake, y_fake = generate_fake_samples(g_model, latent_dim, half_batch)\n",
    "                # update discriminator model\n",
    "                d_loss1 = d_model.train_on_batch(X_real, y_real)\n",
    "                d_loss2 = d_model.train_on_batch(X_fake, y_fake)\n",
    "                # update the generator via the discriminator's error\n",
    "                z_input = generate_latent_points(latent_dim, n_batch)\n",
    "                y_real2 = ones((n_batch, 1))\n",
    "                g_loss = gan_model.train_on_batch(z_input, y_real2)\n",
    "                # summarize loss on this batch\n",
    "                if i % 500 == 0 : \n",
    "                        print('>%d, d1=%.3f, d2=%.3f g=%.3f' % (i+1, d_loss1, d_loss2, g_loss))\n",
    "      \n",
    "\n",
    "\n",
    "# scale images to preferred size\n",
    "def scale_dataset(images, new_shape):\n",
    "        images_list = list()\n",
    "        for image in images:\n",
    "                #print(image.shape)\n",
    "                #print(new_shape)\n",
    "                dim = (new_shape[0],new_shape[1])\n",
    "                # resize with nearest neighbor interpolation\n",
    "                new_image = cv2.resize(image,dim,interpolation = cv2.INTER_NEAREST)\n",
    "                #new_image = np.array(Image.fromarray(image).resize(new_shape, Image.NEAREST))\n",
    "                # store\n",
    "                images_list.append(new_image)\n",
    "        return asarray(images_list)\n",
    "\n",
    "# generate samples and save as a plot and save the model\n",
    "def summarize_performance(status, g_model,d_model, latent_dim, n_samples=25):\n",
    "        # devise name\n",
    "        gen_shape = g_model.output_shape\n",
    "        name = '%03dx%03d-%s' % (gen_shape[1], gen_shape[2], status)\n",
    " \n",
    "        # generate images\n",
    "        X, _ = generate_fake_samples(g_model, latent_dim, n_samples)\n",
    "        # normalize pixel values to the range [0,1]\n",
    "        X = (X - X.min()) / (X.max() - X.min())\n",
    "        # plot real images\n",
    "        square = int(sqrt(n_samples))\n",
    "        for i in range(n_samples):\n",
    "                pyplot.subplot(square, square, 1 + i)\n",
    "                pyplot.axis('off')\n",
    "                pyplot.imshow(X[i])\n",
    "        # save plot to file\n",
    "        value = randint(low=0, high=1000)\n",
    "        filename1 = 'plot_%s.png' % (name + str(value))\n",
    "       \n",
    "        pyplot.savefig(filename1)\n",
    "        pyplot.close()\n",
    "        # save the generator model\n",
    "        filename2 = 'output/model_%s.h5' % (name)\n",
    "        filename3 = 'output/model_disc_%s.h5' % (name)\n",
    "        g_model.save(filename2)\n",
    "        d_model.save(filename3)\n",
    "        print('>Saved: %s and %s and %s' % (filename1, filename2, filename3))\n",
    "\n",
    "# train the generator and discriminator\n",
    "def train(g_models, d_models, gan_models, dataset, latent_dim, e_norm, e_fadein, n_batch):\n",
    "        # fit the baseline model\n",
    "        g_normal, d_normal, gan_normal = g_models[0][0], d_models[0][0], gan_models[0][0]\n",
    "        # scale dataset to appropriate size\n",
    "        gen_shape = g_normal.output_shape\n",
    "        scaled_data = scale_dataset(dataset, gen_shape[1:])\n",
    "        print('Scaled Data', scaled_data.shape)\n",
    "        # train normal or straight-through models\n",
    "        train_epochs(g_normal, d_normal, gan_normal, scaled_data, e_norm[0], n_batch[0])\n",
    "        summarize_performance('tuned', g_normal,d_normal, latent_dim)\n",
    "        # process each level of growth\n",
    "    \n",
    "        for i in range(1, len(g_models)):\n",
    "                # retrieve models for this level of growth\n",
    "                [g_normal, g_fadein] = g_models[i]\n",
    "                [d_normal, d_fadein] = d_models[i]\n",
    "                [gan_normal, gan_fadein] = gan_models[i]\n",
    "                # scale dataset to appropriate size\n",
    "                gen_shape = g_normal.output_shape\n",
    "                scaled_data = scale_dataset(dataset, gen_shape[1:])\n",
    "                print('Scaled Data', scaled_data.shape)\n",
    "                # train fade-in models for next level of growth                     \n",
    "                train_epochs(g_fadein, d_fadein, gan_fadein, scaled_data, e_fadein[i], n_batch[i], True)\n",
    "                summarize_performance('faded', g_fadein,d_fadein, latent_dim)\n",
    "                # train normal or straight-through models\n",
    "                train_epochs(g_normal, d_normal, gan_normal, scaled_data, e_norm[i], n_batch[i])\n",
    "                summarize_performance('tuned', g_normal,d_normal, latent_dim)\n",
    "def load_saved_models(g_model,d_model):\n",
    "             cust = {'PixelNormalization': PixelNormalization, 'MinibatchStdev': MinibatchStdev, 'WeightedSum': WeightedSum}\n",
    "             gen_shape = g_model[0].output_shape\n",
    "             print('input shape -> ' + str(gen_shape))\n",
    "             name = '%03dx%03d-%s' % (gen_shape[1], gen_shape[2], 'tuned')\n",
    "             filename = 'output/model_%s.h5' % (name)\n",
    "             filename2  = 'output/model_disc_%s.h5' % (name)\n",
    "             g_model[0] = load_model(filename, cust)\n",
    "             #d_model[0] = load_model(filename2, cust)\n",
    "             print('loaded : ' + filename)\n",
    "             gen_shape = g_model[1].output_shape\n",
    "\n",
    "             name = '%03dx%03d-%s' % (gen_shape[1], gen_shape[2], 'faded')\n",
    "             filename = 'output/model_%s.h5' % (name)\n",
    "             filename2 = 'output/model_disc_%s.h5' % (name)\n",
    "             g_model[1] = load_model(filename, cust)\n",
    "             #d_model[1] = load_model(filename2, cust)\n",
    "             print('loaded : ' + filename)\n",
    "# number of growth phases, e.g. 6 == [4, 8, 16, 32, 64, 128]\n",
    "n_blocks = 4\n",
    "# size of the latent space\n",
    "latent_dim = 100\n",
    "# define models\n",
    "d_models = define_discriminator(n_blocks)\n",
    "# define models\n",
    "g_models = define_generator(latent_dim, n_blocks)\n",
    "\n",
    "# define composite models\n",
    "gan_models = define_composite(d_models, g_models)\n",
    "\n",
    "#for i in range(1, 4):\n",
    "    #load_saved_models(g_models[i], d_models[i])\n",
    " \n",
    "#print(x)\n",
    "# load image data\n",
    "import numpy as np\n",
    "dataset =  np.load('../../birds2.npy')\n",
    "#load_real_samples('img_align_celeba_128.npz')\n",
    "print('Loaded', dataset.shape)\n",
    "# train model\n",
    "n_batch = [128, 128, 128, 128, 128, 128]\n",
    "# 10 epochs == 500K images per training phase\n",
    "n_epochs = [10, 20, 30, 50, 21, 10]\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "plt.rcParams[\"figure.figsize\"] = (10,10)\n",
    "\n",
    "while True:\n",
    "    train(g_models, d_models, gan_models, dataset, latent_dim, n_epochs, n_epochs, n_batch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
